static/js/components/inputs/StateSelectField.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import _ from 'lodash';
+import iso3166 from 'iso-3166-2';
+import SelectField from './SelectField';
+
+export default class StateSelectField extends React.Component {
+  static propTypes = {
+    profile:                  React.PropTypes.object,
+    updateProfile:            React.PropTypes.func,
+    stateKeySet:              React.PropTypes.array,
+    countryKeySet:            React.PropTypes.array
+  };
+
+  getStateOptions() {
+    const {
+      profile,
+      countryKeySet
+    } = this.props;
+    let options = [];
+    let country = _.get(profile, countryKeySet);
+    if (iso3166.data[country] !== undefined) {
+      options = Object.keys(iso3166.data[country].sub).map(code => ({
+        value: code,
+        label: iso3166.data[country].sub[code].name
```

Comment:
If I'm understanding this correctly should should be able to make this a little more concise with `Object.entries`:

```js
Object.entries(iso3166.data[country].sub).map(([key, value]) => (
  { value: key, label: value.name }
);
```